### PR TITLE
Bug 1202360 - Bring Firefox Account handling modules back to TV system

### PR DESCRIPTION
**Pull request by begeeben**
- Remove apps/system from tv build.
- Merge firefox account feature from apps/system to tv_apps/system.

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d89dacc7d714555618a3ae6577745c97cf49d98a

**Comment by rexboy7** [comment]
Thanks you for working on migrating.
Some concerns below:
- Unit test is broken after moving to smart-system. We may need to repair it or remove it in smart-system temporarily.
- Do we have any policy to align to the latest version of fxa or just leave it as is?

File: build/config/tv/apps-engineering.list [hunk 1]
```diff
@@ -22,7 +22,6 @@ apps/gallery
 apps/keyboard
 apps/music
 apps/pdfjs
-apps/system
```
Comment (rexboy7) [inline build/config/tv/apps-engineering.list]: I guess this deletion raises a GBI test fail. We may need to fix it or just leave apps/system and resolve it in another bug.

File: tv_apps/smart-system/fxa/test/marionette/fxa_screen_flow_test.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  test('should walk screen flow for new user', function () {
+    app.enterEmailNew();
+    app.clickNext();
+    app.clickNext();
+    app.enterPasswordNew();
+    app.clickNext();
+    app.clickDone();
+  });
+
+  test('should walk screen flow for existing user', function () {
```
Comment (rexboy7) [inline tv_apps/smart-system/fxa/test/marionette/fxa_screen_flow_test.js]: It's raising a GIJ test-case fail.

File: tv_apps/smart-system/fxa/test/marionette/fxa_screen_flow_test.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    selectors = FxA.Selectors;
+    app.launch(FxA.UITEST_ORIGIN);
+    app.runUITestMenu();
+  });
+
+  teardown(function() {
+    server.stop();
+  });
+
+  test('should walk screen flow for new user', function () {
```
Comment (rexboy7) [inline tv_apps/smart-system/fxa/test/marionette/fxa_screen_flow_test.js]: It's raising a GIJ test-case fail.

File: tv_apps/smart-system/js/fxa_client.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
+/* -*- Mode: Java; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
```
Comment (rexboy7) [inline tv_apps/smart-system/js/fxa_client.js]: Did you changed the filename? I guess this fire corresponds to `apps/system/fx_accounts_client.js`?

File: tv_apps/smart-system/js/fxa_manager.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
+/* -*- Mode: Java; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
```
Comment (rexboy7) [inline tv_apps/smart-system/js/fxa_manager.js]: Is this fx_accounts_manager.js?

File: tv_apps/smart-system/js/fxa_ui.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
+/* global FxAccountsDialog, FtuLauncher */
```
Comment (rexboy7) [inline tv_apps/smart-system/js/fxa_ui.js]: fx_accounts_u_i.js?

File: build/config/tv/apps-engineering.list (hunk 1 above)
Comment (rexboy7) [inline build/config/tv/apps-engineering.list]: Oh, I see you added IAC for smart-system. So I guess that's why you need to remove the original system app. Let's discuss about this after you're back.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f2181b435d1cfd1385a6f69a2473d2ba11d4ca13

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=639bc69b6247213c3ac51488c4a2c27f05f3c815

**Pull request merged**